integration_tests/src/main/python/cache_test.py

```diff
@@ -28,3 +31,48 @@ def test_passing_gpuExpr_as_Expr():
             .cache()
             .limit(50)
     )
+
+def test_cache_table():
+    spark.sql("CACHE TABLE range5 AS SELECT * FROM range(5)")
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark : spark.sql("select * from range5").limit(5))
+
+gen=[StringGen(nullable=False), DateGen(nullable=False), TimestampGen(nullable=False)]
+@ignore_order
```

Comment:
Just FYI `@ignore_order` does not work if you are going to do the collect yourself.  This might be part of the reason that the test is failing.